src/components/Modal/src/ModalLayer.vue

```diff
@@ -94,6 +94,10 @@ const apiMixin = {
 					vm.currentLayer.state.vnode = undefined; // close modal
 				}
 			},
+
+			async closeAll() {
+				vm.currentLayer.state.vnode = undefined; // close all modals
```

Comment:
Wouldn't this only target the modal on the currentLayer? How are we getting into a state where multiple modals are opened on the same layer?

I'm wondering if we should have a check in place so that new modals cannot be opened on the same layer -- the behavior of continuously opening modals on the same layer is likely contributing towards the need to close all of them.

Reply:
I think we need more context around how multiple modals are being opened at the same time to suggest a better behavior. If one modal is open and another one can open in place of it then I would assume a lot of things can go wrong there. If that's the scenario and this code is being added to fix that problem -- I think we need to change the cause of that problem to begin with, otherwise each engineer is just going to be pulling the rip code to close all modals when that could have a negative experience on customers that are currently using a modal to enter information.

I'm very hesitant to add more code to propagate the idea that multiple modals can even exist. This is a slippery slope of introducing more code, more test, more behaviors that we have to account for rather than pushing for simplification of using modals.